ee/clickhouse/queries/test/test_paths.py

```diff
@@ -2758,6 +2780,215 @@ def test_path_groups_filtering(self):
+        filter = PathFilter(
+            data={
+                "include_event_types": ["$pageview"],
+                "date_from": "2012-01-01 00:00:00",
+                "date_to": "2012-01-02 00:00:00",
+                "path_end_key": "2_/2",
+                "include_recordings": "true",
+            }
+        )
+        _, serialized_actors = ClickhousePathsActors(filter, self.team).get_actors()
+        self.assertEqual([p1.uuid], [actor["id"] for actor in serialized_actors])
+        self.assertEqual(
+            [[]], [actor["matched_recordings"] for actor in serialized_actors],
+        )
+
+    @snapshot_clickhouse_queries
+    @test_with_materialized_columns(["$current_url", "$window_id", "$session_id"])
+    @freeze_time("2012-01-01T03:21:34.000Z")
+    def test_path_recording_with_start_and_end(self):
+        p1 = Person.objects.create(team_id=self.team.pk, distinct_ids=["p1"])
+        events = [
+            _create_event(
+                properties={"$current_url": "/1", "$session_id": "s1", "$window_id": "w1"},
+                distinct_id="p1",
+                event="$pageview",
+                team=self.team,
+                timestamp=timezone.now().strftime("%Y-%m-%d %H:%M:%S.%f"),
+                uuid="11111111-1111-1111-1111-111111111111",
+            ),
+            _create_event(
+                properties={"$current_url": "/2", "$session_id": "s1", "$window_id": "w1"},
+                distinct_id="p1",
+                event="$pageview",
+                team=self.team,
+                timestamp=(timezone.now() + timedelta(minutes=1)).strftime("%Y-%m-%d %H:%M:%S.%f"),
+                uuid="21111111-1111-1111-1111-111111111111",
+            ),
+            _create_event(
+                properties={"$current_url": "/3", "$session_id": "s1", "$window_id": "w1"},
+                distinct_id="p1",
+                event="$pageview",
+                team=self.team,
+                timestamp=(timezone.now() + timedelta(minutes=2)).strftime("%Y-%m-%d %H:%M:%S.%f"),
+                uuid="31111111-1111-1111-1111-111111111111",
+            ),
+        ]
+        _create_all_events(events)
+        self._create_session_recording_event("p1", "s1", timezone.now(), window_id="w1")
+
+        filter = PathFilter(
+            data={
+                "include_event_types": ["$pageview"],
+                "date_from": "2012-01-01 00:00:00",
+                "date_to": "2012-01-02 00:00:00",
+                "path_end_key": "2_/2",
+                "start_point": "/1",
+                "end_point": "/3",
+                "include_recordings": "true",
+            }
+        )
+        _, serialized_actors = ClickhousePathsActors(filter, self.team).get_actors()
+        self.assertEqual([p1.uuid], [actor["id"] for actor in serialized_actors])
+        self.assertEqual(
+            [
+                [
+                    {
+                        "session_id": "s1",
+                        "events": [
```

Comment:
Is this only supposed to have 1 event? What about the start event

Reply:
Right now, it only includes the event of the step in the path that the user clicked on.

I could definitely see us wanting to expand this to include all of the events from the path, but as it stands today, paths can span recordings, so we can't reliably support that feature. And it would be a bit of work to get paths be contained to a recording's session. My vote is we get this feature out, and wait for people to ask for the feature first.